internal/venice-common/src/main/java/com/linkedin/venice/system/store/ControllerClientBackedSystemSchemaInitializer.java

```diff
@@ -97,22 +106,32 @@ public void execute() {
       return;
     }
 
-    try {
-      Pair<String, String> clusterNameAndD2 = admin.discoverCluster(storeName);
-      String currSystemStoreCluster = clusterNameAndD2.getFirst();
+    String storeName = protocolDefinition.getSystemStoreName();
+    Map<Integer, Schema> schemasInLocalResources = Utils.getAllSchemasFromResources(protocolDefinition);
+    D2ServiceDiscoveryResponse discoveryResponse = controllerClient.retryableRequest(
+        DEFAULT_RETRY_TIMES,
+        c -> c.discoverCluster(storeName),
+        r -> ErrorType.STORE_NOT_FOUND.equals(r.getErrorType()));
+    if (discoveryResponse.isError()) {
+      if (ErrorType.STORE_NOT_FOUND.equals(discoveryResponse.getErrorType())) {
+        checkAndMayCreateSystemStore(storeName, schemasInLocalResources.get(1));
+      } else {
+        throw new VeniceException(
+            "Error when discovering system store " + storeName + " after retries. Error: "
+                + discoveryResponse.getError());
+      }
+    } else {
+      String currSystemStoreCluster = discoveryResponse.getCluster();
       if (!currSystemStoreCluster.equals(clusterName)) {
         LOGGER.warn(
-            "The system store for {} already exists in cluster {}, "
-                + "which is inconsistent with the config {} which specifies that it "
-                + "should be in cluster {}. Will abort the initialization routine.",
+            "The system store for {} already exists in cluster {}, which is inconsistent with the config {} which "
```

Comment:
This should be an `error` message, right?
Also I wonder whether we should return some error code for this case, so that Venice Server can choose to abort the startup when this issue happens.

Reply:
Yeah the system store should exist in the system schema cluster. There must be something wrong if it exists in another cluster. I'm ok with throwing an exception to abort server/controller startup when this issue happens.